internal/cluster/cluster.go

```diff
@@ -235,6 +240,101 @@ func (m *Manager) RefreshStatus(ctx context.Context, c *common.Cluster, db *gorm
 	return common.GetClusterFromDB(db, *c.ID, common.UseEagerLoading)
 }
 
+func (m *Manager) hasClusterCompleteInstallation(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster) bool {
+	switch swag.StringValue(cluster.Status) {
+	case models.ClusterStatusInstalled:
+		return true
+	case models.ClusterStatusFinalizing:
+		break
+	default:
+		return false
+	}
+
+	objectName := fmt.Sprintf("%s/%s", cluster.ID, constants.Kubeconfig)
+	exists, err := m.objectHandler.DoesObjectExist(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Warnf("Failed to get object status")
+		return false
+	}
+
+	if !exists {
+		return false
+	}
+
+	status, reason := m.getClusterMonitoringOperatorsStatus(cluster)
+	log.Infof("Monitoring status: %s with reason: %s", status, reason)
+
+	return funk.Contains([]models.OperatorStatus{models.OperatorStatusAvailable, models.OperatorStatusFailed}, status)
+}
+
+func (m *Manager) getClusterMonitoringOperatorsStatus(cluster *common.Cluster) (models.OperatorStatus, string) {
+	numberOfAvailableOperators := 0
+
+	for _, operator := range cluster.MonitoredOperators {
```

Comment:
if olm operator failed cluster should not be "FAILED" and should be in "DEGRADED" state.